### PR TITLE
Adding python tools for debugging and profiling

### PR DESCRIPTION
**Pull request by thomaschabal**
This follows Stéphane Caron's ["Debugging Python code" PR](https://github.com/Simple-Robotics/code-analysis-tools/pull/2).

Additions:
- pdb / pudb for debugging;
- cProfile / snakeviz for profiling.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,22 +28,45 @@ breakpoint set -n functionName
...
+
+```python
+__import__("IPython").embed()
+```
+
+Adding breakpoints in a Python code can be done using `pdb`, with:
+
+```python
+import pdb
+pdb.set_trace() # Execution stops at this point
```
Comment (fabinsch) [inline README.md:58]: what about just using `breakpoint` ?

File: README.md [hunk 2]
```diff
@@ -28,22 +28,45 @@ breakpoint set -n functionName
...
+
+### Quick-and-dirty one-liner
+
+You can spawn a Python interpreter in-context anywhere in your code:
+
+```python
+__import__("IPython").embed()
+```
+
+Adding breakpoints in a Python code can be done using `pdb`, with:
```
Comment (fabinsch) [inline README.md:54]: maybe worth to mention `pdbpp`. Personally, I do `conda install -c conda-forge pdbpp` in my current python env. then is put a `breakpoint` where I want to investigate. 

pdbpp comes with some nice features that make it even more comfortable to debug, like the `sticky` mode . do you think we can integrate this here in your PR directly ?

**Review by fabinsch — COMMENTED** [review]
thanks @thomaschabal, good idea to add the profiling for python as well.

File: README.md (hunk 1 above)
Comment (thomaschabal) [reply README.md:58]: I always used `pdb.set_trace()` and didn't know about `breakpoint` which is indeed simpler to use. I'll replace that part.

File: README.md (hunk 2 above)
Comment (thomaschabal) [reply README.md:54]: I didn't know pdbpp but it seems a great tool. I'll integrate it as well in this PR.

**Pull request closed without merging**